project/yxuan0329/proposal.rst

```diff
@@ -0,0 +1,186 @@
+.. repository.
+
+Basic Information
+=================
+
+.. Create a GitHub repository to host your project and add the URL here.  The
+.. GitHub repository has an 'About' field, in which you are encouraged to write a
+.. simple statement (preferably one sentence) to introduce the project.
+
+R-tree is a tree structure designed for efficient spatial data access. It has 
+been widely used in real-world geographic searching because of its exceptional 
+searching performance. In this project, we aim to serve a R-tree library to 
+provide an efficient search for end-users. 
+
+
+Problem to Solve
+================
+
+.. Describe the problem or the set of problems you want to solve.  Include
+.. necessary background information without making it lengthy.
+
+.. Some points may help you organize the problem description:
+
+.. 1. The field or industry of the problem.
+.. 2. The physics and/or the mathematics behind the problem.
+.. 3. The algorithm or numerical method that should be applied for solving the
+..    problem.
+
+In geographic searching application, the users would expect to swiftly receive 
+a set of searching results upon entering the query. In real-world application, 
+rapid access to geospatial information is crucial for the users, as delays can 
+result in inconveniences and increased costs.  
+
+The fundamental concept behind the R-tree is to encapsulate nearby objects 
+related to the searching target within a minimum bounding rectangle. Our goal 
+is to develop an efficient R-tree library for spatial data retrieval that can 
+address large-scale spatial data searches. 
+
+
+Prospective Users
+=================
+
+.. Describe the users of your software and how they will use it.  It is OK to
+.. combine this section with the previous one (`Problem to solve`_).
+
+This project aims to provide to users who seek to build an efficient 
+multi-dimensional searching service.
+
+
+System Architecture
+===================
+
+.. Analyze how your system takes input, produces results, provide interface, and
+.. performs any other operations.  Describe the system's work flow.  You may
+.. consider to use a flow chart but it is not required.  Specify the constraints
+.. assumed in your system.  Describe the modularization of the system.
+Architecture
+---------
+
+The library would include:
+
+- ``build_tree``: build a r-tree
+- ``search``: search for the available results in r-tree
+- ``insert``: insert a node in r-tree
+- ``delete``: delete a node in r-tree
+
+Datasets
+---------
+
+Randomly generate about 50-100 map data entry by Python for validating the 
+R-tree algorithm can works correctly. 
+
+
+API Description
+===============
+
+.. Show how your system can be programmed.  You are supposed to implement the
+.. system using both C++ and Python.  Describe how a user writes a script in the
+.. system.
+
+The users can include the library by linking in their project. 
+The C++ users can simply link by adding include file.
```

Comment:
Include files cannot be linked.  Include files are included by compilers.  Linkers do not read include files but object files.  Please clarify what do you want.

Reply:
Thanks, I've corrected my proposal.